src/Libraries/RevitNodes/Elements/LinkInstance.cs

```diff
@@ -0,0 +1,425 @@
+﻿using Autodesk.Revit.DB;
+using System;
+using Revit.Elements;
+using System.Collections.Generic;
+using System.Linq;
+using Dynamo.Graph.Nodes;
+using DynamoUnits;
+using View = Revit.Elements.Views.View;
+using System.Text;
+using System.Threading.Tasks;
+using Level = Revit.Elements.Level;
+using RevitServices.Persistence;
+using Autodesk.Revit.UI;
+using Dynamo.Search.SearchElements;
+
+namespace Revit.Elements
+{
+    /// <summary>
+    /// A Revit Link Instance
+    /// </summary>
+    public static class LinkInstance
+    {
+
+        #region Action Nodes
+        /// <summary>
+        /// Retrieves all elements from a link instance on the given level
+        /// </summary>
+        /// <param name="linkInstance">Linked Element Instance </param>
+        /// <param name="level">Linked Element Level </param>
+        /// <returns name="elements[]">List of elements</returns>
+        public static List<Revit.Elements.Element> AllElementsAtLevel(Element linkInstance, Level level)
+        {
+            RevitLinkInstance revitlinkInstance = linkInstance.InternalElement as RevitLinkInstance;
+            Autodesk.Revit.DB.Level revitLevel = level.InternalLevel;
+            ElementId levelId = revitLevel.Id;
+            ElementLevelFilter levelFilter = new ElementLevelFilter(levelId);
+            var linkedAtLevel = new FilteredElementCollector(revitlinkInstance.GetLinkDocument())
+                .WhereElementIsNotElementType()
+                .WherePasses(levelFilter)
+                .Select(el => el.ToDSType(true))
+                .ToList();
+            return linkedAtLevel;
+        }
+
+
+        /// <summary> Retrieves all elements of a given category in a link instance </summary>
+        /// <param name="linkInstance">Revit link Instance</param>
+        /// <param name="category">Element category</param>
+        /// <returns name="linkedElements[]">All elements of the category</returns>
+        public static List<Element> AllElementsOfCategory(Element linkInstance, Revit.Elements.Category category)
+        {
+            RevitLinkInstance revitlinkInstance = linkInstance.InternalElement as RevitLinkInstance;
+            BuiltInCategory bic = (BuiltInCategory)System.Enum.Parse(typeof(BuiltInCategory),
```

Comment:
Why not `BuiltInCategory bic = (BuiltInCategory)category.InternalCategory.Id.Value` ?